ui/lib/kv/addon/components/kv-delete-modal.js

```diff
@@ -57,13 +57,13 @@ export default class KvDeleteModal extends Component {
   }
 
   get deleteOptions() {
-    const { secret, metadata } = this.args;
+    const { secret, metadata, currentVersion } = this.args;
     const isDeactivated = secret.canReadMetadata ? metadata?.currentSecret.isDeactivated : false;
     return [
       {
         key: 'delete-version',
         label: 'Delete this version',
-        description: `This deletes Version ${secret.version} of the secret.`,
+        description: `This deletes Version ${secret.version || currentVersion} of the secret.`,
```

Comment:
If I remember right, they can't access delete-version option if they don't have access to the secret.version (e.g. they don't have metadata read). The idea is they have two delete options: delete a specific version or the current version. This message should only return for the first option: delete a specific version. With that in mind, I don't think this should be an or. We should never be showing this message for deleting the current version. Let me know if that doesn't make sense.

Reply:
It's possible to have `delete` capabilities on a secret that you don't have `read` capabilities on. I caught this during testing that scenario, feel free to play around with this policy to double check: 
```
path "kv/metadata/*" {
  capabilities = ["create", "read", "update", "list"]
}
path "kv/metadata" {
  capabilities = ["list"]
}
path "kv/delete/*" {
    capabilities = ["update"]
}
path "kv/undelete/*" {
    capabilities = ["update"]
}
path "kv/destroy/*" {
    capabilities = ["update"]
}
```

Reply:
But this policy would delete the latest version - so this radio option shouldn't show, correct?

Reply:
Testing with the policy above and this should be switched
<img width="717" alt="Screenshot 2023-08-25 at 8 58 58 AM" src="https://github.com/hashicorp/vault/assets/68122737/24c4d110-22ce-46a5-99fe-52a3eea4b921">